native/arrow_format_nif/src/lib.rs

```diff
@@ -1 +1,195 @@
-rustler::init!("arrow_format_nif", []);
+use rustler::Atom;
+
+mod message;
+mod utils;
+
+use message::record_batch::{Buffer, Compression, FieldNode, RecordBatch};
+use message::schema::field::{Field, Name};
+use message::schema::{Endianness, Feature, Schema};
+use message::{Header, Message, Version};
+
+mod atoms {
+    rustler::atoms! {
+        // TODO Organize atoms in a more systematic order
+
+        // Versions
+
+        v1,
+        v2,
+        v3,
+        v4,
+        v5,
+
+        // Endiannes
+
+        little,
+        big,
+
+        // Layouts
+
+        fixed_primitive,
+        variable_binary,
+        fixed_list,
+        variable_list,
+
+        // Message Types
+
+        schema,
+        record_batch,
+
+        // Features
+
+        unused,
+
+        // Built-ins
+
+        ok,
+        undefined
+    }
+}
+
+/// Returns `:ok` on successful decodes.
+///
+/// This function tests Message's (and its components')
+/// `rustler::types::Decoder` implementation(s), and has only been for testing
+/// purposes.
+#[rustler::nif]
+fn test_decode(_msg: Message) -> Atom {
+    atoms::ok()
+}
+
+/// Returns an example `Message` of a `Schema` or a `RecordBatch`.
+///
+/// This function tests Message's (and its components')
+/// `rustler::types::Encoder` implementation(s), and has only been for testing
+/// purposes.
+#[rustler::nif]
+fn test_encode(msg_type: Atom) -> Message {
+    if msg_type == atoms::schema() {
+        Message {
+            version: Version::V5,
+            header: Header::Schema(Schema {
+                endianness: Endianness::Little,
+                fields: vec![
+                    Field {
+                        name: Name::String("id".to_string()),
```

Comment:
I wonder if you could implement the `From<&str>` trait so that you could do `Name::from("id")`

Reply:
I believe it would be:

```rust
impl From<&str> for Name {
    fn from(value: &str) -> Self {
        Name::String(value.to_string())
    }
}
```